### PR TITLE
update stencil/core and fix #69

### LINKED ISSUE #69: Latest build fails with @stencil/core 1.8.0
This is the counterpart issue to https://github.com/ionic-team/stencil/issues/2015 

I'm submitting a:
[X] bug report

Current behavior:
Updating @stencil/core from 1.7.5 to 1.8.0 breaks compile for stencil-redux

Expected behavior:
Please add some ionic love to stencil-redux and up the core - its 0.18

Steps to reproduce:
Add stencil-redux to a new @stencil/core ^1.8.0 project

Other information:
TypeScript
Import declaration conflicts with local declaration of 'Action'.

./node_modules@stencil\redux\dist\types\global\interfaces.d.ts

import { Action, AnyAction, Store as ReduxStore, Unsubscribe } from 'redux';
export interface Store<S = any, A extends Action = AnyAction> {

### PR DESCRIPTION
**Pull request by stanley85**
Stencil issue https://github.com/ionic-team/stencil/issues/2015
Redux issue https://github.com/ionic-team/stencil-redux/issues/69

This is the adapted solution, dev build works without error - build script runs without error

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "@stencil/redux",
-  "version": "0.1.1",
+  "version": "0.1.2",
   "description": "Stencil Redux - A simple redux-connector for Stencil-built web components",
   "main": "dist/index.js",
-  "module": "dist/esm/es5/index.js",
+  "module": "dist/index.mjs",
```
Comment (stanley85) [inline package.json]: This change was suggested by the compiler

File: package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     "url": "git+https://github.com/ionic-team/stencil-redux.git"
   },
   "devDependencies": {
-    "@stencil/core": "^0.18.0",
+    "@stencil/core": "^1.8.0",
```
Comment (stanley85) [inline package.json]: ^1.8.0 is the first release that is failing with #69

**Comment by dwieeb** [comment]
I think more work is necessary. If you inspect the `dist/` directory, there are no `.js` or `.mjs` files. 

I think this library needs to be completely rewritten for Stencil 1. Not only do we need to fix up stuff like this, we need to abandon the legacy context system and recommend a completely new way of using this library.

**Comment by stanley85** [comment]
Would be awesome to have a syntax like this as soon as redux is available to a stencil project.

`@Store({
   name: 'default',
   action: this.documentChange,
   type: 'redux'
}) document: any = {}; `

Each member decorated with @Store shares the same store. Action is associated with the redux action. @Store Decorator would inherit behaviour from @State. Name could be the name of the reducer.

Having the type as an optional parameter (default = 'redux') could allow to further define custom stores. Maybe a dynamic store for firebase could allow for a simple change like eg.

`@Store({
   name: 'default',
   action: this.documentChange,
   type: 'firestore'
}) document: any = {}; `

Name could be the name of the database document.

**Pull request closed without merging**